io/plugins/eu.esdihumboldt.hale.io.json/src/eu/esdihumboldt/hale/io/json/JsonInstanceReader.java

```diff
@@ -58,17 +86,61 @@ protected IOReport execute(ProgressIndicator progress, IOReporter reporter)
 		progress.begin("Creating " + getDefaultTypeName() + " parser", ProgressIndicator.UNKNOWN);
 
 		try {
-			boolean expectGeoJson = true; // currently defaults to true, no
-											// major difference in functionality
+			boolean expectGeoJson = false;
+			progress.setCurrentTask("Extracting json instances");
+
+			String featureCollection = "FeatureCollection";
+			String typeFeatureCollection = "type";
+
+			String targetField = "@type";
+			ObjectMapper objectMapper = new ObjectMapper();
+
+			try {
+				JsonNode rootNode = objectMapper.readTree(new File(getSource().getLocation()));
```

Comment:
This reads the whole Json document into memory, that is definitely not something we want to do.
Ideal would be to only use the streaming API, but that is hard to do because there is no guaranteed order the fields will appear in. So the compromise I settled for was only reading a whole object/feature into memory at a time.